### PR TITLE
feat: use closure instead of class

### PR DESCRIPTION
**Pull request by FezVrasta**
image

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -65,174 +144,98 @@ export default class Popper {
...
-    // Run `onDestroy` modifier methods
-    this.state.orderedModifiers.forEach(
-      ({ onDestroy, enabled }) => enabled && onDestroy && onDestroy(this.state)
-    );
-  }
+    return {
+      forceUpdate,
+      update,
+      destroy,
+    };
```
Comment (atomiks) [inline src/index.js]: I think this should be stored as a variable and methods are accessed like `instance.update()` - this allows public accessors to monkey-patch the methods if necessary

Comment (FezVrasta) [reply src/index.js]: Good idea! Done

File: README.md [hunk 2]
```diff
@@ -104,12 +104,14 @@ import {
...
+  defaultModifiers: [
+    detectOverflow,
+    preventOverflow,
+    computeStyles,
+    applyStyles,
+  ],
+});
 ```
 
 Now you can use the `Popper` class with _only_ the features you want. For
```
Comment (atomiks) [inline README.md]: Says `class`

**Comment by FezVrasta** [comment]
I removed the default export, since there can be problems when mixing default and named exports on legacy module systems. So now it's `import { createPopper, popperGenerator } from '@popperjs/core';`

**Comment by atomiks** [comment]
I like the capitalized version. Seems fien to me, although I kind of do like having a default export a bit more.

**Comment by FezVrasta** [comment]
I'm avoiding the mixed export because in some systems you will end up with stuff like this:

```
const createPopper = require('@popperjs/core').default;
```

and:
```
const { default: createPopper, popperGenerator } = require('@popperjs/core');
```

**Comment by atomiks** [comment]
Yeah tippy has that :\, I don't really care about the CJS version because tree-shaking doesn't work anyway (in webpack). It's a last-resort so ugly things like that don't matter too much imo

**Comment by FezVrasta** [comment]
What about UMD? I'm not even sure it can support mixed exports.

**Comment by atomiks** [comment]
What I did was attach it to the function namespace, it's far cleaner to use for UMD:

```js
createPopper.popperGenerator = popperGenerator;
```

**Comment by FezVrasta** [comment]
Isn't a named export clearer? 🤨

**Comment by atomiks** [comment]
I do think `Popper.createPopper` is pretty clean. 

In Tippy it would've been `Tippy.tippy` which is weird since it's a generic constructor instead of one that creates a single instance. In this case it works better 👍

**Comment by FezVrasta** [comment]
Ok cool, then let's stick with named exports 🤘

**Pull request merged**